Add Recipe for sentence-navigation.el

angelic-sedition opened the pull request:
[sentence-navigation.el](https://github.com/angelic-sedition/emacs-sentence-navigation) provides sentence navigation commands (like the default forward-sentence) that are abbreviation aware (abbreviations won't end a sentence) as well as sentence text objects that can be used with evil.

purcell commented:
Thanks! Quick feedback:

`sn-` is too short a prefix. Just use `sentence-navigation-`! :-)  And don't use `/`, particularly for public symbols -- it's worth following the [elisp coding conventions](https://www.gnu.org/software/emacs/manual/html_node/elisp/Coding-Conventions.html).

Re. the `(require 'evil)` block -- just because `evil` can be loaded doesn't mean that the user wants to use it. Instead, use an `(eval-after-load 'evil ...)` block.

Hope that helps,

-Steve

angelic-sedition commented:
Thank you for the feedback. I've implemented your second suggestion except using `with-eval-after-load` (hopefully that's okay).

As for the prefix length, I'd like to keep it as short as possible. Short prefixes don't seem to be particularly uncommon (e.g. sp for smart-parens and yas for yasnippet). Would a three letter prefix be acceptable? If not, would `sentence-nav-` be acceptable? It kind of makes me cringe and seems excessive to have commands as long as `sentence-navigation-evil-backward-sentence-end`.

purcell commented:
Very short prefixes are bad because they can lead to clashes with unrelated packages. So `sentence-nav-` would be a better choice here, but characters are cheap, and longer command names are no problem for users, who probably aren't going to type `ss-evil-backward-sentence-end` either! "sentence-navigation-" would be far from the longest prefix I've seen.

Also, since the package is all about sentences, you can probably name symbols such that they don't need a second mention of "sentence", e.g. `sentence-navigation-evil-backward-end`, which isn't too cumbersome. :-)

angelic-sedition commented:
Yeah it's not such a problem practically. I personally just don't like looking at really long command names. Anyway, thank you for looking over it; I have now implemented your suggestion and changed the package prefix to `sentence-nav-`.

purcell commented:
Great - merged!
